pkg/transport/process/downloaders/downloaders_suite_test.go

```diff
@@ -0,0 +1,216 @@
+	Expect(keyring.AddAuthConfig(testenv.Addr, credentials.AuthConfig{
+		Username: testenv.BasicAuth.Username,
+		Password: testenv.BasicAuth.Password,
+	})).To(Succeed())
+	ociCache = cache.NewInMemoryCache()
+	var err error
+	ociClient, err = ociclient.NewClient(logr.Discard(), ociclient.WithKeyring(keyring), ociclient.WithCache(ociCache))
+	Expect(err).ToNot(HaveOccurred())
+
+	uploadTestComponent()
+}, 60)
+
+var _ = AfterSuite(func() {
+	Expect(testenv.Close()).To(Succeed())
+})
+
+func uploadTestComponent() {
+	ctx := context.TODO()
+	fs := memoryfs.New()
+
+	localOciBlobRes := createLocalOciBlobRes(fs)
+	imageRes := createImageRes(ctx)
+	imageIndexRes := createImageIndexRes(ctx)
+
+	ociRepo := cdv2.NewOCIRegistryRepository(testenv.Addr+"/test/downloaders", "")
+	repoCtx, err := cdv2.NewUnstructured(
+		ociRepo,
+	)
+	Expect(err).ToNot(HaveOccurred())
+
+	localCd := cdv2.ComponentDescriptor{
+		ComponentSpec: cdv2.ComponentSpec{
+			ObjectMeta: cdv2.ObjectMeta{
+				Name:    "github.com/component-cli/test-component",
+				Version: "0.1.0",
+			},
+			Provider: cdv2.InternalProvider,
+			RepositoryContexts: []*cdv2.UnstructuredTypedObject{
+				&repoCtx,
+			},
+			Resources: []cdv2.Resource{
+				localOciBlobRes,
+				imageRes,
+				imageIndexRes,
+			},
+		},
+	}
+
+	manifest, err := cdoci.NewManifestBuilder(ociCache, ctf.NewComponentArchive(&localCd, fs)).Build(ctx)
+	Expect(err).ToNot(HaveOccurred())
+
+	ociRef, err := cdoci.OCIRef(*ociRepo, localCd.Name, localCd.Version)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(ociClient.PushManifest(ctx, ociRef, manifest)).To(Succeed())
+
+	cdresolver := cdoci.NewResolver(ociClient)
+	actualCd, err := cdresolver.Resolve(ctx, ociRepo, localCd.Name, localCd.Version)
+	Expect(err).ToNot(HaveOccurred())
+
+	testComponent = *actualCd
+}
+
+func createLocalOciBlobRes(fs vfs.FileSystem) cdv2.Resource {
+	Expect(fs.Mkdir(ctf.BlobsDirectoryName, os.ModePerm)).To(Succeed())
+
+	dgst := digest.FromBytes(localOciBlobData)
+	blobfile, err := fs.Create(ctf.BlobPath(dgst.String()))
+	Expect(err).ToNot(HaveOccurred())
+
+	_, err = blobfile.Write(localOciBlobData)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(blobfile.Close()).To(Succeed())
+
+	localOciBlobAcc, err := cdv2.NewUnstructured(
+		cdv2.NewLocalFilesystemBlobAccess(
+			dgst.String(),
+			"text/plain",
+		),
+	)
+	Expect(err).ToNot(HaveOccurred())
+
+	localOciBlobRes := cdv2.Resource{
+		IdentityObjectMeta: cdv2.IdentityObjectMeta{
+			Name:    "local-oci-blob",
+			Version: "0.1.0",
+			Type:    "plain-text",
+		},
+		Relation: cdv2.LocalRelation,
+		Access:   &localOciBlobAcc,
+	}
+
+	return localOciBlobRes
+}
+
+func createImageRes(ctx context.Context) cdv2.Resource {
+	imageRef = testenv.Addr + "/test/downloaders/image:0.1.0"
+
+	manifest, desc, err := testutils.UploadTestManifest(ctx, ociClient, imageRef)
+	Expect(err).ToNot(HaveOccurred())
+
+	// TODO: currently needed to fill the cache. remove from test, also from ociclient unit test
```

Comment:
TODO?

Reply:
the exact same todo is in 2 other places of the coding if i recall correct. Therefore, would fix this for all of the 3 occurences from main branch